hydradx/notebooks/Omnipool/vault/ImpermanentLossAnalysis.ipynb

```diff
@@ -0,0 +1,225 @@
+      "text/plain": [
+       "<Figure size 1440x360 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "steps = 101\n",
+    "# price_min = 0.2\n",
+    "# price_max = 5\n",
+    "for price_min, price_max in [(0.2, 5), (0.5, 2)]:\n",
+    "    step_factor = (price_max / price_min) ** (1 / (steps - 1))\n",
+    "    price_move_range = [price_min * step_factor ** i for i in range(steps)]\n",
+    "    events = []\n",
+    "    weight_factors = [0.0001, 0.01, 0.02, 0.035, 0.05]\n",
+    "\n",
+    "\n",
+    "    for w, weight in enumerate(weight_factors):\n",
+    "        assets = {\n",
+    "            'HDX': {'usd price': 0.05, 'weight': 0.05},\n",
+    "            'USD': {'usd price': 1, 'weight': 0.8 - weight},\n",
+    "            'TKN': {'usd price': 1, 'weight': weight}\n",
```

Comment:
Shouldn't this have all 3 tokens, i.e. shouldn't this have TKN2 as well?

Reply:
You're right, this is making the weights wrong